File: python/src/Aged_brie.py
from src.Item import Item
from src.updateable import updateable

class AgedBrie(Item, updateable):
    def __init__(self, name, sell_in, quality):
        Item.__init__(self, name, sell_in, quality)
    
    def getSell_in(self):
        return self.sell_in
    
    def getQuality(self):
        return self.quality
    
    def updateQuality(self):
        if self.sell_in <= 0:
            self.quality += 2
        else:
            self.quality += 1
        self.maxQuality()
        self.sell_in -= 1

    def maxQuality(self):
        if self.quality >= 50:
            self.quality = 50

File: java/src/main/java/edu/poniperro/gildedrose/AgedBrie.java
package edu.poniperro.gildedrose;

public class AgedBrie extends Item implements Updateable {

	public AgedBrie(String name, int sell_in, int quality) {
		super(name, sell_in, quality);
	}

	public void updateQuality() {
		if (getSell_in() <= 0) {
			setQuality(getQuality() + 2);
		} else {
			setQuality(getQuality() + 1);
		}
		checkQuality();
		setSell_in();
	}

	private void checkQuality() {
		if (getQuality() >= 50) {
			setQuality(50);
		}
	}
}